File: Module_Base_De_Donnees/Tp 2eme annee/Tp NoSQL/Tp 1/tp1.js
use("gst_stg");

//question 1
db.Stagiaires.find();

//question 2
db.Examens.find();

//question 3
db.Stagiaires.find({}, { NomS: 1, PrenomS: 1, _id: 0 });

//question 4
db.Examens.find({}, { NumE: 1, Date: 1 });

//question 5
db.Stagiaires.find().sort({ NomS: -1, _id: 0 });

//question 6
db.Examens.find({ Type: "p" }, { _id: 0 }).sort({ Date: 1 });

//question 7
db.PasserExamen.find({ NumS: { $eq: "S01" } }, { NumE: 1, Note: 1 }).sort({
  Note: 1,
});

//question 8
db.Examens.find({ Salle: { $in: ["A2", "A3"] } }, {});

//question 9
db.PasserExamen.find(
  { $or: [{ NumS: "S01" }, { Note: { $gte: 15 } }] },
  { NumE: 1, Note: 1 }
);

//question 10
db.PasserExamen.find({ Note: { $gt: 15 } }, { NumS: 1 });

//question 11
db.PasserExamen.find({ NumS: "S01" }, {}).limit(2);

//question 12
db.Stagiaires.find({ NomS: { $regex: /^A/ } }, {});

//question 13
db.Examens.find().count();

//question 14
db.Examens.find({ Salle: "B5" }).count();

//question 15
db.PasserExamen.find({ NumE: { $in: ["E01", "E02", "E03"] } }, { NumS: 1 });

File: Module_Base_De_Donnees/Tp 2eme annee/Tp NoSQL/Tp pyMongo/Tp 1/tp1.py
from pymongo import MongoClient
import re

conx = MongoClient("mongodb://localhost:27017/")

db = conx["gst_stg"]

stagiaires = db["Stagiaires"]

examens = db["Examens"]

passerExamen = db["PasserExamen"]

# Q1
for s in stagiaires.find() :
    print(s)

# Q2
for e in examens.find() :
    print(e)

# Q3
for s in stagiaires.find({}, {"nom": 1, "prenom": 1}) :
    print(s)

# Q4
for e in examens.find({}, {"NumE": 1, "Date": 1}) :
    print(e)

# Q5
for s in stagiaires.find().sort("NomS", -1) :
    print(s)

# Q6
for e in examens.find({"Type": "p"}).sort("Date") :
    print(e)

# Q7
for p in passerExamen.find({"NumS": "S01"}, {"_id": 0, "NumE": 1, "Note": 1, "NumS": 0, "Salle": 0}).sort("Note") :
    print(p)

# Q8
for p in passerExamen.find({"Salle": {"$in": ["A2", "A3"]}}) :
    print(p)

# Q9
for p in passerExamen.find({"$and": [{"NumS": "S01"}, {"Note": {"$gte": 15}}]}) :
    print(p)

# Q10
for p in passerExamen.find({"Note": {"$gt": 15}}, {"_id": 0, "NumE": 0, "Note": 0, "NumS": 1, "Salle": 0}) :
    print(p)

# Q11
for p in passerExamen.find({"NumS": "S01"}, {}).limit(2) :
    print(p)

# Q12
for s in stagiaires.find({"NomS": {"$regex": "^A"}}) :
    print(s)

# or
pattern = re("/^A/")
for s in stagiaires.find({"NomS": {"$regex": pattern}}) :
    print(s)

# Q13
print("Le Nombre Des Examens Est: " + examens.count_documents())

# Q14
print("Le Nombre Des Examens Passer Dans La Salle B5 Est: " + examens.count_documents({"Salle": "B5"}))

# Q15
for p in passerExamen.find({"NumE": {"$in": ["E01", "E02", "E03"]}}, {"_id": 0, "NumE": 0, "Note": 0, "NumS": 1, "Salle": 0}) :
    print(p)